pytest_mh/_private/fixtures.py

```diff
@@ -249,26 +249,51 @@ def _teardown(self) -> None:
         if errors:
             raise Exception(errors)
 
-    def _collect_artifacts(self, host: MultihostHost) -> None:
+    def _artifacts_dir(self) -> str | None:
         """
-        Collect test artifacts that were requested by the multihost configuration.
+        Return test artifact directory or ``None`` if no artifacts should be
+        stored.
 
-        :param host: Host object where the artifacts will be collected.
-        :type host: MultihostHost
+        :return: Artifact directory or ``None``.
+        :rtype: str | None
         """
         if self._skipped:
-            return
+            return None
 
         dir = self.request.config.getoption("mh_artifacts_dir")
         mode = self.request.config.getoption("mh_collect_artifacts")
         if mode == "never" or (mode == "on-failure" and self.data.outcome != "failed"):
-            return
+            return None
 
         name = self.request.node.name
         for c in list('":<>|*?'):
             name = name.replace(c, "-")
 
-        host.collect_artifacts(f"{dir}/{name}")
+        return f"{dir}/{name}"
+
+    def _collect_artifacts(self, host: MultihostHost) -> None:
+        """
+        Collect test artifacts that were requested by the multihost configuration.
+
+        :param host: Host object where the artifacts will be collected.
+        :type host: MultihostHost
+        """
+        path = self._artifacts_dir()
+        if path is None:
+            return
+
+        host.collect_artifacts(path)
+
+    def _flush_logs(self) -> None:
+        """
+        Write log messages produces by current test case to a file or clear
+        them if not artifacts should be generated.
```

Comment:
Minor grammar fixes
~~~
Write log messages produced by current test case to a file, or clear them if no artifacts should be generated.
~~~

Reply:
Fixed, thank you.